### ISSUE TITLE
Using `for` in CSS attribute selector throws error.

### ISSUE THREAD

**Issue by abdulsattar** [salesforce/lwc#3470]
### Description
`for` is a valid attribute, used heavily in `label` tags. CSS attribute selectors that target `for` throw errors in LWC.
#### Steps to Reproduce

Repro:
https://stackblitz.com/edit/salesforce-lwc-xjayej?file=src%2Fmodules%2Fx%2Fapp%2Fapp.css&title=LWC%20playground

Code
```css
label[for="expand"] {
  color: red;
}
```

#### Expected Results
No error is thrown.

#### Actual Results
Invalid usage of attribute selector "for". "for" is a template directive and therefore not supported in css rules.

**Comment by SaurabhMulay999** [comment]
@abdulsattar 
As For in LWC reserved as template directive so you can use data-input-id
 
label[data-input-id='expand']{
  color: red;
}

**Comment by nolanlawson** [comment]
Is this disallowed for the same reason we disallow `[id="..."]`? `for` references an ID (IDref), so it would be broken in synthetic shadow.

Related: #2571

**Comment by abdulsattar** [comment]
You're right, I think it was disallowed for that reason. All the points that were raised in #2571 remain valid for this bug as well.

**Comment by nolanlawson** [comment]
Interestingly, we don't do this for `aria-labelledby` ([repro](https://stackblitz.com/edit/salesforce-lwc-xfqecp?file=src/modules/x/app/app.css)). This might be a separate bug from #2571.
